app/src/main/java/com/example/una/adapters/CategoryAdapter.java

```diff
@@ -69,12 +71,20 @@ public ViewHolder(@NonNull View itemView) {
             super(itemView);
             ivCategoryImage = (ImageView) itemView.findViewById(R.id.ivCategoryImage);
             tvCategoryName = (TextView) itemView.findViewById(R.id.tvCategoryName);
-            itemView.setOnClickListener(this);
+            itemView.setOnClickListener(new CategoryClickListener());
         }
 
-        @Override
-        public void onClick(View v) {
-            // TODO display list of charities in that category
+        class CategoryClickListener implements View.OnClickListener {
+            @Override
+            public void onClick(View view) {
+                int categoryItemPosition = getAdapterPosition();
+                // getting the category name based on which the user clicked on
+                Category category = (Category) categories.get(categoryItemPosition);
+                // fire an intent to an activity with a recycler view populated with a list of charities matching each category
+                Intent categoryList = new Intent(view.getContext(), CharitySearchListActivity.class);
+                categoryList.putExtra("query", category.categoryName);
```

Comment:
Is this intentional? This would retrieve a list of charities as if the user had entered a query string of the category name, but we want to bring up a list of charities filtered by category, so I think you'll need to add another parameter for `categoryId` that you put when `fetchCharities` is called in `CharitySearchListActivity.java`

Reply:
https://charity.3scale.net/docs/data-api/reference#working-with-the-api
![image](https://user-images.githubusercontent.com/43116228/62728006-945cf680-b9cf-11e9-85ad-2cd0fbbe677f.png)